libsplinter/src/admin/service/shared.rs

```diff
@@ -881,11 +1002,45 @@ impl AdminServiceShared {
             .make_disband_request_circuit_proposal(circuit_id, requester, requester_node_id)
             .map_err(|err| ServiceError::UnableToHandleMessage(Box::new(err)))?;
 
-        self.check_connected_peers_payload_disband(
-            &circuit_proposal.get_circuit_proposal().get_members(),
-            payload,
-            message_sender,
-        )
+        let members = circuit_proposal
+            .get_circuit_proposal()
+            .get_members()
+            .iter()
+            .map(|member| {
+                match circuit_proposal
```

Comment:
This seems as though the this could also be some kind of `try_as_token()` fn, it's repeated a lot here, with little difference